orttraining/orttraining/test/python/orttraining_test_python_bindings.py

```diff
@@ -0,0 +1,244 @@
+    pt_model.train()
+    # setting all initial weights to zero
+    if zero_flag:
+        with torch.no_grad():
+            for param in pt_model.parameters():
+                param.zero_()
+
+    x = torch.randn(batch_size, input_size, device=device)
+    onnx_model = _get_onnx_model(pt_model, (x,))
+
+    return pt_model, onnx_model
+
+
+def _get_onnx_model(torch_model, model_inputs):
+    model_outputs = torch_model(*model_inputs)
+    if isinstance(model_outputs, torch.Tensor):
+        model_outputs = [model_outputs]
+    dynamic_axes = {}
+    input_names = []
+    output_names = []
+    for i, model_input in enumerate(model_inputs):
+        input_name = f"input-{i}"
+        input_names.append(input_name)
+        dynamic_axes[input_name] = {}
+        for dim_idx in range(len(model_input.shape)):
+            dynamic_axes[input_name].update({dim_idx: f"{input_name}_dim{dim_idx}"})
+
+    for i, model_output in enumerate(model_outputs):
+        output_name = f"output-{i}"
+        output_names.append(output_name)
+        dynamic_axes[output_name] = {}
+        for dim_idx in range(len(model_output.shape)):
+            dynamic_axes[output_name].update({dim_idx: f"{output_name}_dim{dim_idx}"})
+
+    f = io.BytesIO()
+    torch.onnx.export(
+        torch_model,
+        model_inputs,
+        f,
+        input_names=input_names,
+        output_names=output_names,
+        opset_version=14,
+        do_constant_folding=False,
+        training=torch.onnx.TrainingMode.TRAINING,
+        dynamic_axes=dynamic_axes,
+        export_params=True,
+        keep_initializers_as_inputs=False,
+    )
+    return onnx.load_model_from_string(f.getvalue())
+
+
+def _create_training_models():
+    # Given
+    device = "cuda"
+    batch_size, input_size, hidden_size, output_size = 64, 784, 500, 10
+    _, onnx_model = _get_models(device, batch_size, input_size, hidden_size, output_size)
+
+    # Build the onnx model with loss
+    simple_model = SimpleModelWithCrossEntropyLoss()
+    with onnxblock.onnx_model(onnx_model) as accessor:
+        _ = simple_model(onnx_model.graph.output[0].name)
+        eval_model = accessor.eval_model
+
+    optimizer = onnxblock.optim.AdamW()
+    with onnxblock.onnx_model() as accessor:
+        _ = optimizer(simple_model.parameters())
+        optimizer_model = accessor.model
+
+    return simple_model, onnx_model, optimizer_model, eval_model
+
+
+def test_train_step():
+    # Initialize Models
+    simple_model, onnx_model, _, _ = _create_training_models()
+    trainable_params, non_trainable_params = simple_model.parameters()
+
+    # Generating random data for testing.
+    inputs = torch.randn(64, 784).numpy()
+    labels = torch.randint(high=10, size=(64,), dtype=torch.int32).numpy()
+    forward_inputs = OrtValueVector()
+    forward_inputs.reserve(2)
+    forward_inputs.push_back(OrtValue.ortvalue_from_numpy(inputs)._ortvalue)
+    forward_inputs.push_back(OrtValue.ortvalue_from_numpy(labels)._ortvalue)
```

Comment:
are users expected to pass in `_ortvalue`? Can this be handled internally by our function? Users should only need to be aware of the python `OrtValue`.